AFMReader/general_loader.py

```diff
@@ -0,0 +1,89 @@
+"""Switchboard for input files."""
+
+from pathlib import Path
+
+from AFMReader import asd, gwy, ibw, jpk, spm, stp, top, topostats
+from AFMReader.logging import logger
+
+logger.enable(__package__)
+
+
+# pylint: disable=too-few-public-methods
+class LoadFile:
+    """
+    Class to handle the general loading of an AFM file.
+
+    Parameters
+    ----------
+        filepath : Path
+            Path to the AFM image.
+        channel : str
+            Channel to extract from the AFM image.
+    """
+
+    def __init__(self, filepath: Path, channel: str):
+        """
+        Initialise the general LoadFile class with a filepath and channel.
+
+        Parameters
+        ----------
+        filepath : Path
+            Path to the AFM image.
+        channel : str
+            Channel to extract from the AFM image.
+        """
+        self.filepath = filepath
+        self.channel = channel
+        self.suffix = filepath.suffix
+
+    def load(self):  # noqa: C901
```

Comment:
```suggestion
    def load(self) -> tuple[npt.NDArray | str, float | None]:  # noqa: C901
```

something like this for a type hint? though it's a bit of a special case